### ISSUE TITLE
v3: Can't add very first repo

### ISSUE THREAD

**Issue by marckhouzam** [helm/helm#6521]
The acceptance-tests started failing after #6492 (had to do a git bisect to figure that out).

The reason is that it is no longer possible to add the very first repo when starting from an empty environment as shown below:
```
> export XDG_CONFIG_HOME=/tmp/test/config
> mkdir -p ${XDG_CONFIG_HOME}

> $GOPATH/src/helm.sh/helm/bin/helm repo add gitlab https://charts.gitlab.io
Error: open /tmp/test/config/helm/repositories.yaml: no such file or directory

> /bin/rm -rf ${XDG_CONFIG_HOME}
```

The reason is that the locking of repository file expects the directory of the `repositories.yaml` file to exist.

I first thought of ignoring the error if it was a missing file error, but then I thought that the race the locking logic is fixing could also happen if two `helm repo add` try to create the `repositories.yaml` file concurrently.  I don't know how to fix that...

**Comment by karuppiah7890** [comment]
After seeing the above PR recently, I was planning to use locks for #6140 issue. Thanks for the heads up on what kind of issues I can expect if I implement things using locks

**Comment by hickeyma** [comment]
I have pushed a fix in #6555.

**Comment by karuppiah7890** [comment]
This can be merged as the PR #6555 is merged

**Comment by hickeyma** [comment]
Closing as #6555 merged. Thanks for heads up @karuppiah7890.
